Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1

```diff
@@ -396,11 +397,25 @@ function Invoke-CodeGen {
     Invoke-Task -name $MyInvocation.MyCommand.Name -task {
         $tool = (Join-Path $toolsPath 'EdFi.Ods.CodeGen')
         $repositoryRoot = (Get-RepositoryRoot $implementationRepo).Replace($implementationRepo, '')
-
-        if ($IncludePlugins) {
+              
+        if (-not [string]::IsNullOrWhiteSpace($ExtensionPaths)) {           
+            Foreach ($extensionPath in $ExtensionPaths) {        
+                if (![System.IO.Directory]::Exists($extensionPath)) {
+                    throw "Unable to find extension Location project path at location $extensionPath ."
+                }
+            }
+        }
+       
+        if ($IncludePlugins -and $ExtensionPaths.Length -eq 0 ) {
             & $tool -r $repositoryRoot -e $Engine -IncludePlugins | Write-Host
         }
-        else {
+        elseif ($IncludePlugins -and $ExtensionPaths.Length -gt 0 ) {
+            & $tool -r $repositoryRoot -e $Engine -IncludePlugins --ExtensionPaths $ExtensionPaths | Write-Host
+        }
+        elseif (-not  $IncludePlugins -and $ExtensionPaths.Length -gt 0 ) {
+            & $tool -r $repositoryRoot -e $Engine --ExtensionPaths $ExtensionPaths | Write-Host
+        }
+        elseif (-not  $IncludePlugins -and $ExtensionPaths.Length -eq 0 ) {
```

Comment:
these if statements are still pretty hard to read. would something like this be easier to read and maintain?
https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/blob/694ffafd7b628ae70bdb2f684895e2bfe02ac258/logistics/scripts/modules/packaging/create-package.psm1#L132-L149